src/Build/Construction/ProjectRootElement.cs

```diff
@@ -712,6 +720,18 @@ internal ProjectExtensionsElement ProjectExtensions
         internal string LastDirtyReason
             => _dirtyReason == null ? null : String.Format(CultureInfo.InvariantCulture, _dirtyReason, _dirtyParameter);
 
+        /// <summary>
+        /// Initialize an in-memory, empty ProjectRootElement instance that can be saved later.
+        /// The ProjectRootElement will not be marked dirty.
+        /// Uses the global project collection.
```

Comment:
Is the global project collection appropriate? Shouldn't we use the projectcollection that the project we're in mid-evaluation for uses?

Reply:
i used ProjectCollection.GlobalProjectCollection because the method Cretae() called in Evalutor uses it.

yes, this makes sense. I will check if this works

Reply:
it worked. thanks! done